### PR TITLE
 chore: move update-private-narhash to script 

### PR DESCRIPTION
**Pull request by zimbatm**
The issue with the current dev-flake setup is that `builtins.getFlake`
is not available in Nix, unless flakes are enabled. So this idea
of using flake-compat doesn't work.

Since the project is exporting modules, it doesn't matter. Replace the
top-level default.nix flake-compat with a simple attrset.

### CODE REVIEW

File: dev/update-private-narhash [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+# Used to update the private dev flake hash reference.
+set -euo pipefail
+
+cd "$(dirname "$0")"
+
+echo "Updating $PWD/private.narHash" >&2
+
+nix flake lock ./private
```
Comment (Mic92) [inline dev/update-private-narhash:9]:
```suggestion
nix --extra-experimental-flags 'flakes nix-command' flake lock ./private
```

File: dev/update-private-narhash [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+nix hash path ./private > ./private.narHash
```
Comment (Mic92) [inline dev/update-private-narhash:10]:
```suggestion
nix --extra-experimental-flags 'flakes nix-command' hash path ./private > ./private.narHash
```

**Comment by Mic92** [comment]
@mergify queue

File: dev/update-private-narhash (hunk 2 above)
Comment (zowoq) [reply dev/update-private-narhash]: features, not flags.

https://github.com/nix-community/srvos/commit/a6ac04bf8ac513d8e4a506d7d69c0f161e4b528c

**Pull request merged**